pointcloud/src/main/scala/geotrellis/pointcloud/raster/ept/DEMRasterSource.scala

```diff
@@ -102,16 +102,124 @@ case class DEMRasterSource(
+          val s1y = pc.getY(c)
+
+          val v2x = pc.getX(b)
+          val v2y = pc.getY(b)
+          val v2z = pc.getZ(b)
+          val s2x = pc.getX(a)
+          val s2y = pc.getY(a)
+
+          val v3x = pc.getX(c)
+          val v3y = pc.getY(c)
+          val v3z = pc.getZ(c)
+          val s3x = pc.getX(b)
+          val s3y = pc.getY(b)
+
+          val determinant =
+          (v2y - v3y) * (v1x - v3x) + (v3x - v2x) * (v1y - v3y)
+
+          val ymin =
+            math.min(v1y, math.min(v2y, v3y))
+          val ymax =
+              math.max(v1y, math.max(v2y, v3y))
+
+          val scanrow0 = math.max(math.ceil((ymin - eymin) / h - 0.5), 0)
+          var scany = eymin + scanrow0 * h + h / 2
+          while (scany < eymax && scany < ymax) {
+            // get x at y for edge
+            var xmin = Double.MinValue
+            var xmax = Double.MaxValue
+
+            if(s1y != v1y) {
+              val t = (scany - v1y) / (s1y - v1y)
+              val xAtY1 = v1x + t * (s1x - v1x)
+
+              if(v1y < s1y) {
+                // Lefty
+                if(xmin < xAtY1) { xmin = xAtY1 }
+              } else {
+                // Righty
+                if(xAtY1 < xmax) { xmax = xAtY1 }
+              }
+            }
+
+            if(s2y != v2y) {
+              val t = (scany - v2y) / (s2y - v2y)
+              val xAtY2 = v2x + t * (s2x - v2x)
+
+              if(v2y < s2y) {
+                // Lefty
+                if(xmin < xAtY2) { xmin = xAtY2 }
+              } else {
+                // Righty
+                if(xAtY2 < xmax) { xmax = xAtY2 }
+              }
+            }
+
+            if(s3y != v3y) {
+              val t = (scany - v3y) / (s3y - v3y)
+              val xAtY3 = v3x + t * (s3x - v3x)
+
+              if(v3y < s3y) {
+                // Lefty
+                if(xmin < xAtY3) { xmin = xAtY3 }
+              } else {
+                // Righty
+                if(xAtY3 < xmax) { xmax = xAtY3 }
+              }
+            }
+
+            val scancol0 = math.max(math.ceil((xmin - exmin) / w - 0.5), 0)
+            var scanx = exmin + scancol0 * w + w / 2
+            while (scanx < exmax && scanx < xmax) {
+              val col = ((scanx - exmin) / w).toInt
+              val row = ((eymax - scany) / h).toInt
+              if(0 <= col && col < cols &&
+                 0 <= row && row < rows) {
+
+                val z = {
+
+                  val lambda1 =
+                  ((v2y - v3y) * (scanx - v3x) + (v3x - v2x) * (scany - v3y)) / determinant
+
+                  val lambda2 =
+                  ((v3y - v1y) * (scanx - v3x) + (v1x - v3x) * (scany - v3y)) / determinant
+
+                  val lambda3 = 1.0 - lambda1 - lambda2
+
+                  lambda1 * v1z + lambda2 * v2z + lambda3 * v3z
+                }
+
+                tile.setDouble(col, row, z)
+              }
+
+              scanx += w
+            }
+
+            scany += h
+          }
```

Comment:
How do you feel about moving this code into some `TriangleRasterizer`?

Reply:
That is possible, but we'd want to check performance of that?  This rasterizer pulls some amount of state from the surrounding scope.  This was done originally because of the use of the half-edge mesh.  I'd be concerned about increasing the call times at all of a function that will be run hundreds of thousands of times.  We declared this function inline in the DelaunayRasterizer, which was tuned for performance, so I followed that model.  We can come back and test whether it makes a difference later, but for now, I'd say leave it in place.

Reply:
I don't follow your implications, but I'll merge it as it is.